packages/keystone/apolloErrorFormatter.js

```diff
@@ -96,130 +97,222 @@ function _patchKnownErrorCases (error, result) {
-    const originalErrorClassNames = originalError ?  _getClassList(originalError) : []
-    // NOTE 1: If GQLError is thrown at ASYNC field hook, it will be wrapped like following:
-    // GraphQLError + Wrapper: {
-    //      originalError (Error): {
-    //          errors: [
-    //              GQLError
-    //          ]
-    //      }
-    // }
-    // So we need to unwrap this specific scenario
-    // NOTE 2: addFieldValidationError will do the same wrap, but its originalError will have a `name` prop,
-    // and it also will have additional classNames on top or Error
-    // We can use this info, but I've decided to be more straightforward:
-    // Obtain inner GQLError. If not found - use default behaviour
-    let isInnerGQLErrorFound = false
-    if (errorClassNames.includes('GraphQLError') && errorClassNames.includes('Wrapper') && originalErrorClassNames.includes('Error')) {
-        const innerError = _extractInnerGQLError(originalError)
-        if (innerError) {
-            isInnerGQLErrorFound = true
-            const formattedInnerError = safeFormatError(innerError, hideInternals, applyPatches)
-            Object.assign(result, formattedInnerError)
-            // Note: Inner error extraction, if no originals -> originalError = self
-            originalError = get(formattedInnerError, 'originalError', formattedInnerError)
-            result.originalError = originalError === formattedInnerError ? formattedInnerError : safeFormatError(originalError, hideInternals, false)
-        }
+    // NOTE(pahaz): we want to extract internal knex error messages from violates constraint cases
+    const hasDBUniqConstrain = originalError && originalError?.message?.includes('duplicate key value violates unique constraint')
+    const hasDBCheckConstrain = originalError && originalError?.message?.includes('violates check constraint')
+    if (hasDBUniqConstrain || hasDBCheckConstrain) {
+        _handleKnexError(extensions, originalError)
+        return
     }
+}
 
-    if (!isInnerGQLErrorFound && originalError) {
-        result.originalError = safeFormatError(originalError, hideInternals, false)
+function _safeFormatErrorRecursion (errorIn, hideInternals = false, applyPatches = true, _isRecursionCall = false) {
+    const error = _ensureError(errorIn)
+    const errorCName = error?.constructor?.name
+    const extensions = {}
+    const originalError = error?.originalError
+    const originalErrorCName = originalError?.constructor?.name
+    const result = {}
+
+    // [1] base error fields: name, message, stack
+    const pickKeys1 = (hideInternals) ? ['message', 'name'] : ['message', 'name', 'stack']
+    Object.assign(result, pick(error, pickKeys1))
+
+    // [2] base graphql fields: locations, path
+    //  - `locations` - array of { line, column } locations
+    //  - `path` - array describing the JSON-path
+    const pickKeys3 = ['locations', 'path']
+    Object.assign(result, pick(error, pickKeys3))
```

Comment:
Where are the `pickKeys2`? :)

Reply:
Ok, just hardcode it without any variable